aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -380,11 +384,27 @@ module aptos_framework::account {
     }
 
     /// A resource account is used to manage resources independent of an account managed by a user.
-    public fun create_resource_account(source: &signer, seed: vector<u8>): (signer, SignerCapability) {
-        let addr = create_resource_address(&signer::address_of(source), seed);
-        let signer = create_account_unchecked(copy addr);
-        let signer_cap = SignerCapability { account: addr };
-        (signer, signer_cap)
+    public fun create_resource_account(source: &signer, seed: vector<u8>): (signer, SignerCapability) acquires Account {
+        let resource_addr = create_resource_address(&signer::address_of(source), seed);
+        let resource = if (exists_at(resource_addr)) {
+            let account = borrow_global<Account>(resource_addr);
+            assert!(
+                option::is_none(&account.signer_capability_offer.for),
```

Comment:
This seems to presume that a pre-existing resource account has a `CapabilityOffer` installed? But that was not the case in the previous version of the code (red lines 383-387).

I'm not sure I understand what edge case this is protecting for.

Reply:
Oh, it's because I haven't read below...

Reply:
historically, we only checked to see if the account was created, it couldn't be recreated and the signer could not be given away twice.

with this change, we actually introduce a small window for bad behavior, where someone could have deployed a testnet smart contract with the intent of never being able to access that account again. every single testnet resource account that has been allocated prior to this change, will be recreateable at least once... I reckon there's nothing we can do here as we don't have any data to suggest when the account was created.